### ISSUE TITLE
Always loads previous broken state in saved file until configuration deleted

### ISSUE THREAD

**Issue by bryanparadis** [lukehaas/RunJS#416]
Finally figured out the problem was that Run.js persistently reopened code that had an error in it even if I:

1. Cleared all code in tab and saved
2. Cleared all code in tab and saved as another filename
3. Closed the tab and pressed new tab
4. Uninstalled Run.js completely

Deleting the configuration file at `./Library/Application Support/runjs/config.json` solved the issue.

**Comment by bryanparadis** [comment]
#### Config:
```
{
	"autoeval": true,
	"transpile": true,
	"showUndefined": false,
	"loopProtection": true,
	"lineNumbers": true,
	"invisibles": false,
	"matchingLines": true,
	"activeLine": false,
	"lineWrap": true,
	"vimkeys": true,
	"userId": "b8233bc7-9aab-4131-9379-ff6d19e8f67e",
	"fontSize": "14",
	"font": "DejaVu Sans Mono",
	"formatOnRun": false,
	"closeBrackets": true,
	"width": 1800,
	"height": 1074,
	"abc": false,
	"envVars": [],
	"track": true,
	"prettierConfig": {
		"singleQuote": true
	},
	"babelPlugins": {
		"decoratorsPlugin": false,
		"functionSentPlugin": false,
		"doExpressionsPlugin": false,
		"asyncDoExpressionsPlugin": false,
		"pipelineOperatorPlugin": false,
		"partialApplicationPlugin": false,
		"throwExpressionsPlugin": false,
		"recordAndTuple": false
	},
	"babelPresets": {
		"typeScriptPreset": true,
		"reactPreset": false,
		"envPreset": true
	},
	"statementResults": true,
	"autoScroll": true,
	"updateAvailable": false,
	"tabs": {
		"c7386b82-7c2a-492a-8ef7-0038e9a1511c": {
			"saveFilePath": "/Users/b/broken",
			"arrangement": {
				"direction": "row",
				"first": "editor",
				"second": "output",
				"splitPercentage": 55
			}
		},
		"order": [
			"c7386b82-7c2a-492a-8ef7-0038e9a1511c"
		],
		"activeTabID": "c7386b82-7c2a-492a-8ef7-0038e9a1511c"
	},
	"tabBarForSingleTab": true,
	"autocomplete": false,
	"diagnostics": false,
	"hideMenuBar": false,
	"showTranspiled": false,
	"verticalLayout": true,
	"theme": "dracula",
	"hasUpdated": false,
	"firstOpen": false,
	"currentVersion": "2.3.0",
	"downloadComplete": true,
	"x": 0,
	"y": 44,
	"analytics": true
}
```

#### Random assortment of doodles:
```
var example = {
    foo: "foo1",
    foo2: "foo2",
    foo3: "foo3",
    foo: "foo4",
};

example.constructor.name


      var countdown = 2;
      var x = setInterval(function() {
        if (countdown < 1)
          window.close();
        else
          --countdown;
        document.getElementById("countdown").innerHTML = countdown;
      }, 1000);


//function blah() {
//  stdout = executeShellCommand('command');
//  
//  if (stdout.trim() === "") return [];
//
//  return stdout.split("\n").map((command) => ({
//    name: command,
//    description: 'test description'
//  }));
//
//}
//
//blah()

//function build_subcommand (name, description) {
//  return {
//    name: name,
//    description: description
//  };
//}
//

//const { exec } = require("child_process");
//exec("/opt/dev/bin/dev for-completion list-commands", (error, stdout, stderr) => {
//  
//if (stdout.trim() === "") return [];
//
//commands = stdout.split("\n")
//  .map((command) => ({
//  name: command,
//  description: 'test description',
//}));
//  
//blah = { name: "dev",
//         subcommands: [commands],
//}
//     
//
//console.log(blah)
//  
//});

```

**Comment by lukehaas** [comment]
The issue here was due to this line in your code `window.close();` causing the app to crash.
I've now resolved this in the latest release 2.3.1
